### PR TITLE
Fallback locales

### PR DESCRIPTION
**Pull request by madtrick**
### Summary

Documentation for the new fallback behaviour in locales.

### CODE REVIEW

File: apiary/cma.apib [hunk 1]
```diff
@@ -752,7 +752,25 @@ An asset can only be archived when it's not published.
 # Group Locales
 
 Locales allow the definition of translated content for both assets and entries. A locale
-consists mainly of a name and a locale code.
+consists mainly of the following properties:
+
+- `name`, a human readable identifier for a locale. For example: "British English"
+- `code`, an identifier that will be used to differentiate translated content in the API responses. For example: 'en-GB'
+- `fallback_code`, the code of the locale to use in case there is no translated content for the requested one. For example: 'en-US'
```
Comment (hlabas) [inline apiary/cma.apib]: I'd add "or `null` for no fallback." at the end of this bullet point.

File: apiary/cma.apib [hunk 2]
```diff
@@ -752,7 +752,25 @@ An asset can only be archived when it's not published.
...
+On those cases the API will, by default, return content for the default locale in the space (if there's any content for it). You can override this behaviour
+by setting a custom `fallback_code` in your locale. If there's no content for a locale but it exists for the fallback one then the content delivery api
+will return its content.
+
+Some considerations about fallback locales:
+
+- If you don't want a locale to fallback to anything set its `fallback_code` property to `null`.
+- You can create fallback chains. Setups where for example `de-CH` (Swiss German) fallbacks to `de-AT` (Austrian German) and `de-AT` fallbacks itself to `de-DE` (German German) are
+perfectly valid. If a locale in the chain doesn't have content the API will try with the next one.
+- Your fallback chain can't contain cycles. A fallback chain where `de-CH` fallbacks to `de-AT` and `de-AT` fallbacks to `de-CH` won't be allowed.
```
Comment (hlabas) [inline apiary/cma.apib]: I believe we should provide a short set of explanations about what is triggering a fallback to happen, i.e. the "empty" concept to be clear about these compared to `null` or `""` values.
Something along the lines of:

> A field's value will fallback only if no value was provided upon update in (link to Create/update an entry  [PUT]). This means that setting the value to `null` or `""` for the locale will result in `null` or `""` to be returned by a future Content Delivery API call.

File: apiary/cma.apib [hunk 3]
```diff
@@ -832,6 +852,9 @@ This endpoint deletes an existing locale. Please note that it is **not** possibl
 from this action! Every content that was stored for that specific locale gets deleted and
 cannot be recreated by creating the same locale again.
 
+Deleting a locale that it's used as a fallback is not allowed. You would have to first ensure that
```
Comment (hlabas) [inline apiary/cma.apib]: that it's used >> that is used

**Comment by madtrick** [comment]
@hlabas PTAL

File: apiary/cma.apib [hunk 4]
```diff
@@ -779,7 +779,28 @@ An asset can only be archived when it's not published.
...
+will return its content.
+
+Some considerations about fallback locales:
+
+- If you don't want a locale to fallback to anything set its `fallbackCode` property to `null`.
+- You can create fallback chains. Setups where for example `de-CH` (Swiss German) fallbacks to `de-AT` (Austrian German) and `de-AT` fallbacks itself to `de-DE` (German German) are
+perfectly valid. If a locale in the chain doesn't have content the API will try with the next one.
+- Your fallback chain can't contain cycles. A fallback chain where `de-CH` fallbacks to `de-AT` and `de-AT` fallbacks to `de-CH` won't be allowed.
+- The system will only use the content of the fallback locale when the requested one is not present in the entry or asset (i.e. you request 'en-US' but the entry has only content for 'de-DE').
+This means tha values like `null` or an empty string (`""`) will not cause the system to use the value of the fallback locale.
```
Comment (hlabas) [inline apiary/cma.apib]: tha >> that

**Comment by hlabas** [comment]
LGTM, thanks Farruco

File: apiary/cma.apib (hunk 4 above)
Comment (madtrick) [inline apiary/cma.apib]: :facepalm:

File: apiary/cma.apib (hunk 1 above)
Comment (madtrick) [inline apiary/cma.apib]: done

File: apiary/cma.apib (hunk 2 above)
Comment (madtrick) [inline apiary/cma.apib]: done

File: apiary/cma.apib (hunk 3 above)
Comment (madtrick) [inline apiary/cma.apib]: done

File: apiary/cma.apib (hunk 4 above)
Comment (madtrick) [inline apiary/cma.apib]: fixed

File: apiary/_partials/delivery-and-preview-resources.apib [hunk 5]
```diff
@@ -30,12 +30,20 @@ Each space has a name, a set of locales, and some system metadata.
...
+              "name": "English",
+              "fallbackCode": null
             },
             {
               "code": "tlh",
               "default": false,
-              "name": "Klingon"
+              "name": "Klingon",
+              "fallbackCode": "en-US"
+            }
```
Comment (realityking) [inline apiary/_partials/delivery-and-preview-resources.apib]: Tests are failing because of the missing comma on this line.

**Pull request merged**